### PR TITLE
mobilevit model v0.01

### PR DESCRIPTION
**Pull request by XiaoheYou**

### CODE REVIEW

File: easycv/models/layers/dropout.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        p: probability of an element to be zeroed. Default: 0.5
+        inplace: If set to ``True``, will do this operation in-place. Default: ``False``
+
+    Shape:
+        - Input: :math:`(N, *)` where :math:`N` is the batch size
+        - Output: same as the input
+
+    """
+
+    def __init__(self,
```
Comment (wenmengzhou) [inline easycv/models/layers/dropout.py:26]: 这种layer为什么不直接用nn.Dropout？ 同其他layer

File: easycv/models/classification/mobilevit_v2/base_cls.py [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+            self.dilate_l4 = True
+            self.dilate_l5 = True
+        elif output_stride == 16:
+            self.dilate_l5 = True
+
+        self.model_conf_dict = dict()
+        self.logger = get_root_logger()
+
+    @classmethod
+    def add_arguments(cls, parser: argparse.ArgumentParser):
```
Comment (wenmengzhou) [inline easycv/models/classification/mobilevit_v2/base_cls.py:49]: 为什么需要这个接口？ 移除

File: easycv/models/classification/mobilevit_v2/mobilevit_v2.py [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+from .modules import MobileViTBlockv2 as Block
+
+
+@MODELS.register_module
+class MobileViTv2(BaseEncoder):
+    """
+    This class defines the MobileViTv2 architecture
+    """
+
+    def __init__(self, *args, **kwargs) -> None:
```
Comment (wenmengzhou) [inline easycv/models/classification/mobilevit_v2/mobilevit_v2.py:22]: 请参考其他backbone把主要参数在__init__函数中声明

File: easycv/models/classification/mobilevit_v2/mobilevit_v2.py [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+        )
+
+        # check model
+        self.check_model()
+
+        # weight initialization
+        self.reset_parameters(opts=opts)
+
+    @classmethod
+    def add_arguments(
```
Comment (wenmengzhou) [inline easycv/models/classification/mobilevit_v2/mobilevit_v2.py:124]: 移除

File: easycv/models/classification/mobilevit_v2/base_cls.py [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+                module_name,
+                'Params',
+                round(layer_param / 1e6, 3),
+                'MACs',
+                round(layer_macs / 1e6, 3),
+            ))
+            print('-' * 67)
+        return input, overall_params, overall_macs
+
+    def profile_model(
```
Comment (wenmengzhou) [inline easycv/models/classification/mobilevit_v2/base_cls.py:195]: 这些不需要

File: easycv/models/classification/mobilevit_v2/mobilevit_v2.py [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+                attn_norm_layer=getattr(
+                    opts, 'model.classification.mitv2.attn_norm_layer',
+                    'layer_norm_2d'),
+                dilation=self.dilation,
+            ))
+
+        return nn.Sequential(*block), input_channel
+
+
+def MobileViTv2_postprocess(out):
```
Comment (wenmengzhou) [inline easycv/models/classification/mobilevit_v2/mobilevit_v2.py:254]: 这个可以抽象到forward_test接口里，放到主干网络里

File: easycv/models/layers/__init__.py [hunk 7]
```diff
@@ -0,0 +1,82 @@
+#
```
Comment (jhuang1207) [inline easycv/models/layers/__init__.py:1]: 这个layers目录是干啥的？看起来就是把官方代码抄了一份过来？这个完全没有按照EasyCV本身的设计理念进行开发。这个大的代码结构的调整，必须有详细的设计文档，以及调整的理由

File: easycv/models/layers/activation/__init__.py [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+        default=0.1,
+        type=float,
+        help='Negative slope in leaky relu function',
+    )
+
+    return parser
+
+
+__all__ = [
+    'GELU',
```
Comment (jhuang1207) [inline easycv/models/layers/activation/__init__.py:75]: relu，sigmoid等等，原util文件夹下本来就有，activation乃至layers文件夹存在的意义是什么？

File: easycv/models/layers/normalization/__init__.py [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+        '--model.normalization.adjust-bn-momentum.final-momentum-value',
+        default=1e-6,
+        type=float,
+        help='Min. momentum in batch normalization layer',
+    )
+
+    return parser
+
+
+__all__ = [
```
Comment (jhuang1207) [inline easycv/models/layers/normalization/__init__.py:93]: 这些norm为啥还要单独实现

**Review by jhuang1207 — CHANGES_REQUESTED** [review]
看起来代码提交者对easyCV的整体设计风格没有足够的认识，建议
1 了解EasyCV代码结构并彻底重构mobilevit的实现
2 增加benchmark，提供复现精度和速度的文档
3 增加测试代码
